packages/maskbook/src/components/InjectedComponents/NFT/NFTAvatar.tsx

```diff
@@ -242,10 +242,18 @@ export async function getNFT(address: string, tokenId: string) {
     return {
         amount: order ? new BigNumber(getOrderUnitPrice(order) ?? 0).toFixed() : '0',
         name: asset.assetContract.name,
-        symbol: order?.paymentTokenContract?.symbol ?? '',
+        symbol: order?.paymentTokenContract?.symbol ?? 'ETH',
         image: asset.imageUrl ?? asset.imagePreviewUrl ?? '',
     }
 }
+
+export function useNFT(address: string, tokenId: string) {
+    return useAsyncRetry(async () => {
+        const nft = await getNFT(address, tokenId)
+        return nft
+    }, [address, tokenId, getNFT])
```

Comment:
```suggestion
    }, [address, tokenId])
```

`getNFT` is a static function, no need to add to dependencies array.